### PR TITLE
 RUN-3014 deprecate rvm send 

### PR DESCRIPTION
**Pull request by datamadic**
Switch out the `send` call for the rvm bus in favor of the typed `publish` call. 

[tests well](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/592349de5114ac10ca6db44c)

### CODE REVIEW

**Comment by wenjunche** [comment]
👍

File: src/browser/rvm/runtime_initiated_topics/app_assets.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    };
+
+    /**
+     *  Notifies all observers of relevant alias info! (or lack thereof)
+     *
+     */
+    private handleInfoResponse (sourceUrl: string, dataObj: any) {
+        _.mapObject(this.pendingRequests[sourceUrl], (requestedAliasCallbackArray: any, alias: string) => {
+            const aliasInResponse = _.findWhere(dataObj.result, {
+                'alias': alias
```
Comment (joneit) [inline src/browser/rvm/runtime_initiated_topics/app_assets.ts]: `{ alias }`

File: src/browser/rvm/runtime_initiated_topics/app_assets.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+import {rvmMessageBus, AppAssetsGetList} from '../rvm_message_bus';
+import * as log from '../../log';
+import * as  _ from 'underscore';
+
+interface CallbackObj {
+    successCB: Function;
+    failureCB: Function;
+}
```
Comment (datamadic) [inline src/browser/rvm/runtime_initiated_topics/app_assets.ts]: yes, will edit

File: src/browser/rvm/runtime_initiated_topics/app_assets.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+    };
+
+    // Returns bool which indicates whether this is the 1st request for sourceUrl - then we actually need to send it to RVM
+    private addPendingRequest (sourceUrl: string, assetAlias: string, successCB: Function, failureCB: Function) {
+        const pendingCBObj = {
+            successCB: successCB,
+            failureCB: failureCB
+        };
```
Comment (datamadic) [inline src/browser/rvm/runtime_initiated_topics/app_assets.ts]: yes, will edit

File: src/browser/rvm/rvm_message_bus.ts [hunk 4]
```diff
@@ -86,104 +215,127 @@ class RVMMessageBus extends EventEmitter  {
      *
      **/
     // TODO: the type of the data here is to be defined in RUN-2947
-    public send(topic: string, data: any, callback: Function, timeToLiveInSeconds: number) {
+    // public send(topic: string, data: any, callback: Function, timeToLiveInSeconds: number) {
```
Comment (datamadic) [inline src/browser/rvm/rvm_message_bus.ts]: Its not yet 'defunct' per se, there are still parts of the code that use it I did not want to change over everything just yet. In a later PR this is removed

File: src/browser/rvm/runtime_initiated_topics/app_assets.ts (hunk 1 above)
Comment (datamadic) [inline src/browser/rvm/runtime_initiated_topics/app_assets.ts]: yes, will edit

File: src/browser/rvm/rvm_message_bus.ts [hunk 5]
```diff
@@ -129,6 +139,15 @@ export interface System extends RvmMsgBase {
     sourceUrl: string;
 }
 
+// topic: applicaiton-events -----
```
Comment (datamadic) [inline src/browser/rvm/rvm_message_bus.ts]: yes, will edit

File: src/browser/rvm/rvm_message_bus.ts [hunk 6]
```diff
@@ -129,6 +139,15 @@ export interface System extends RvmMsgBase {
...
+type EventType = 'started'| 'closed' | 'ready' | 'run-requested' | 'crashed' | 'error' | 'not-responding' | 'out-of-memory';
+type ApplicaitonEventTopic = 'application-event';
```
Comment (datamadic) [inline src/browser/rvm/rvm_message_bus.ts]: yes, will edit

File: src/browser/rvm/rvm_message_bus.ts (hunk 4 above)
Comment (datamadic) [inline src/browser/rvm/rvm_message_bus.ts]: sorry, yes, this is an error I will edit

File: src/browser/rvm/utils.ts [hunk 7]
```diff
@@ -26,25 +22,13 @@ interface SendToRVMOpts {
...
-        };
 
         // Make sure there is a connection with RVM
-        if (!rvmBus) {
+        if (!rvmMessageBus) {
             return reject(new Error('Connection with RVM is not established'));
         }
 
-        const messageSent = rvmBus.send(topic, payload, (rvmResponse: any) => {
+        const messageSent = rvmMessageBus.publish(Object.assign(opts, {timeToLive: 1000}), (rvmResponse: any) => {
```
Comment (datamadic) [inline src/browser/rvm/utils.ts]: yes I like this, will edit

**Comment by datamadic** [comment]
@joneit @HarsimranSingh updated, have a look

[re-ran tests](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59273e40998bc04e20cc6c84) look good

**Pull request merged**